dev-requirements.txt

```diff
@@ -7,7 +7,6 @@ kubernetes==25.3.0
 flake8==3.9.0
 fsspec==2024.3.1
 s3fs==2024.3.1
-google-api-core
```

Comment:
note: removed because this is actually pinned to `google-api-core==2.18.0` in L47 below and running `pip install -r dev-requirements.txt` fails if the same package is listed twice in this file.